File: example/ApiTest/Go/chats/1_create_chat.go
package main

import (
  "fmt"
  "strings"
  "net/http"
  "io"
)

func main() {

  url := "http://127.0.0.1/api/v1/chats"
  method := "POST"

  payload := strings.NewReader(`{
    "dataset_ids": ["8a85ab34ad5311ef98b00242ac120003"],
    "name":"my test chat"
}`)

  client := &http.Client {
  }
  req, err := http.NewRequest(method, url, payload)

  if err != nil {
    fmt.Println(err)
    return
  }
  req.Header.Add("Content-Type", "application/json")
  req.Header.Add("Authorization", "Bearer ragflow-hjNzA4ODI4YWM5MTExZWY5YzUyMDI0Mm")

  res, err := client.Do(req)
  if err != nil {
    fmt.Println(err)
    return
  }
  defer res.Body.Close()

  body, err := io.ReadAll(res.Body)
  if err != nil {
    fmt.Println(err)
    return
  }
  fmt.Println(string(body))
}

File: example/ApiTest/Python/chats/1_create_chat.py
import requests
import json

url = "http://127.0.0.1/api/v1/chats"

payload = json.dumps({
  "dataset_ids": [
    "8a85ab34ad5311ef98b00242ac120003"
  ],
  "name": "my test chat"
})
headers = {
  'Content-Type': 'application/json',
  'Authorization': 'Bearer ragflow-hjNzA4ODI4YWM5MTExZWY5YzUyMDI0Mm'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)
